operator/config/metrics/monitor.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: metrics-proxy
+  namespace: system
+  labels:
+    control-plane: controller-manager
+    app.kubernetes.io/name: servicemonitor
+    app.kubernetes.io/instance: controller-manager-metrics-monitor
+    app.kubernetes.io/component: metrics
+    app.kubernetes.io/created-by: workspaces
+    app.kubernetes.io/part-of: workspaces
+    app.kubernetes.io/managed-by: kustomize
+spec:
+  endpoints:
+    - interval: 15s
+      scheme: http
+      path: /metrics
+      port: metrics
+      authorization:
+        credentials:
+          key: token
+          name: metrics-reader
```

Comment:
the service account creation automatically creates a token secret which name is not predictable. How are you going to make sure there is a secret named metrics-reader?

Reply:
The secret gets generated in `operator/config/metrics/kustomization.yaml`.

Reply:
We also write the name of this generated secret into the service monitor via a replacement in `operator/config/default/kustomization.yaml`:

```yaml
replacements:
- source:
    fieldPath: metadata.name
    kind: Secret
    name: metrics-reader
  targets:
  - fieldPaths:
    - spec.endpoints.*.authorization.credentials.name
    select:
      group: monitoring.coreos.com
      kind: ServiceMonitor
      name: metrics-proxy
```